src/Hpack/Render.hs

```diff
@@ -127,6 +127,7 @@ renderPackage settings alignment existingFieldOrder sectionsFieldOrder Package{.
           makeVersion [1,22] <$ guard hasReexportedModules
         , makeVersion [2,0]  <$ guard hasSignatures
         , makeVersion [2,0] <$ guard hasGeneratedModules
+        , makeVersion [2,2] <$ guard (hasCxxParams sect)
```

Comment:
I think right now `internal-libraries` are not taken into consideration.  What we probably want to do is use `libraryCabalVersion` in `internalLibsCabalVersion` similar to what we already with `executableCabalVersion` and `executablesCabalVersion`.